lib/shopwareStoreCommander.js

```diff
@@ -126,21 +127,82 @@ module.exports = class ShopwareStoreCommander {
 
     /**
      * @param {Object} plugin
-     * @param {String} filePath
+     * @param {String} version
+     * @param {Array} changelogs
+     * @param {Array} compatibleShopwareVersions
      * @return {Object}
      */
-    async uploadPluginBinary(plugin, filePath) {
-        const binaryName = filePath.split(/(\\|\/)/g).pop();
-        this.logEmitter.emit('info', `Uploading binary ${binaryName} for plugin ${plugin.name}...`);
+    async createPluginBinary(plugin, version, changelogs, compatibleShopwareVersions) {
+        this.logEmitter.emit('info', `Creating binary version ${version} of plugin ${plugin.name}...`);
 
-        const { data, headers } = await getPostDataFromFile(filePath);
-        const res = await this.client.post(`plugins/${plugin.id}/binaries`, data, { headers });
+        const accountData = await this.getAccountData();
+        const { data: newBinary } = await this.client.post(
+            `producers/${accountData.producerId}/plugins/${plugin.id}/binaries`,
+            {
+                version,
+                changelogs,
+                softwareVersions: compatibleShopwareVersions,
+            },
+        );
 
-        // Add the binary info to the plugin
-        plugin.binaries = res.data;
+        const matchingBinaryIndex = plugin.binaries.findIndex(existingBinary => existingBinary.id === newBinary.id);
+        if (matchingBinaryIndex === -1) {
+            plugin.binaries.push(newBinary);
+        } else {
+            plugin.binaries[matchingBinaryIndex] = newBinary;
+        }
         plugin.latestBinary = plugin.binaries[plugin.binaries.length - 1];
 
-        return plugin;
+        return newBinary;
+    }
+
+    /**
+     * @param {Object} plugin
+     * @param {Object} binary
+     * @param {Array} changelogs
+     * @param {Array} compatibleShopwareVersions
+     * @param {Boolean} licenseCheckRequired
+     * @return {Object}
+     */
+    async updatePluginBinary(plugin, binary, changelogs, compatibleShopwareVersions, licenseCheckRequired) {
+        this.logEmitter.emit('info', `Updating binary version ${binary.version} of plugin ${plugin.name}...`);
+
+        const accountData = await this.getAccountData();
+        const { data: updatedBinary } = await this.client.put(
+            `producers/${accountData.producerId}/plugins/${plugin.id}/binaries/${binary.id}`,
+            {
+                changelogs,
+                softwareVersions: compatibleShopwareVersions,
+                licenseCheckRequired,
+                // ionCube encryption is no longer supported
+                ionCubeEncrypted: false,
```

Comment:
Can't we just skip sending the parameter then?

Reply:
The request sent by the Shopware account app contained this in its payload. Without it I got an HTTP 400.